tests/Kron.py

```diff
@@ -37,14 +39,111 @@
 
 #-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
 
-class KronPhotometryTestCase(unittest.TestCase):
+def makeGalaxy(width, height, flux, a, b, theta, dx=0.0, dy=0.0, xy0=None, xcen=None, ycen=None):
```

Comment:
I guess that's right.  I looked at master to see what the original was,
when I should have looked at the origin of my branch.

OK, I will clean up before merging.  Sorry to put you through the trouble.

If you are saying that I should merge to master before asking for review, I
can do that.  That has not been part of the procedure in the past, but I'm
happy to do that.  It is rebasing against my own branch which I found not
very useful.

On Thu, Feb 19, 2015 at 6:01 AM, John Swinbank <notifications@github.com>
wrote:

> In tests/Kron.py
> <https://github.com/lsst/meas_extensions_photometryKron/pull/1#discussion_r24988770>
> :
>
> > @@ -37,14 +39,111 @@
> >
> >  #-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
> >
> > -class KronPhotometryTestCase(unittest.TestCase):
> > +def makeGalaxy(width, height, flux, a, b, theta, dx=0.0, dy=0.0, xy0=None, xcen=None, ycen=None):
>
> I think there's something more complex going on here, actually. To
> elaborate:
>
> The makeGalaxy function was introduced by Paul in cc2925d
> <https://github.com/lsst/meas_extensions_photometryKron/commit/cc2925d056657e997545f532451917060aa85b02>
> several months ago:
>
> $ git log --all --format="%h, %an, %ar: %s" -SmakeGalaxy
> 4ed8e0a, pgee, 3 weeks ago: DM-982 Convert kron algorithm to meas_base
> fe56449, pgee, 2 weeks ago: Additional fixups:
> cc2925d, Paul Price, 7 months ago: tests: add test for forced photometry under rotation, translation, scaling
>
> Note that when Paul added made that change, he didn't just add the
> makeGalaxy function, but *moved* code from KronPhotometryTestCase into
> that function.
>
> However, cc2925d
> <https://github.com/lsst/meas_extensions_photometryKron/commit/cc2925d056657e997545f532451917060aa85b02>
> is not part of the history of your branch:
>
> $ git branch -r --contains cc2925d
>   origin/HEAD -> origin/master
>   origin/master
>   origin/tickets/DM-1783
>
> Instead, you seem to have introduced makeGalaxy yourself in 4ed8e0a
> <https://github.com/lsst/meas_extensions_photometryKron/commit/4ed8e0ac6eaf6011e0418b945e53550fa4adeb6a>.
> This is (as far as git is concerned) completely independent of Paul's code
> -- it looks as though you copied and pasted it, or something. However, you
> only copied the additions from Paul's commit (ie, the creation of the
> makeGalaxy function), not the removal of the code from elsewhere in that
> file. That means you've ended up with a whole bunch of duplication.
>
> So what does that mean? First, you have to clean this up before you can
> merge it; it can't wait to go on a separate ticket. Secondly, it's pretty
> confusing for both you and Paul to have apparently spontaneously written
> exactly the same code; you need to clean up history so that that doesn't
> appear to have been the case.
>
> My suggestion would be to rebase your changes on top of master before
> merging. As part of that rebase, you'll be adding Paul's cc2925d
> <https://github.com/lsst/meas_extensions_photometryKron/commit/cc2925d056657e997545f532451917060aa85b02>
> to the history of your branch. When you try to apply your own 4ed8e0a
> <https://github.com/lsst/meas_extensions_photometryKron/commit/4ed8e0ac6eaf6011e0418b945e53550fa4adeb6a>
> on top of it, you'll probably find that it conflicts. You'll need to
> resolve the conflict, effectively by deleting your copy of makeGalaxy and
> just using the one Paul wrote.
>
> It's interesting to note that if you'd cleaned up your branch and rebased
> it against master before asking for a review, you'd already have had to fix
> this problem and I wouldn't have had to spend time questioning it and
> figuring out what happened. It's a nice example of the benefits of
> pre-review rebasing, as we were discussing just the other day!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lsst/meas_extensions_photometryKron/pull/1/files#r24988770>
> .
>